py/sentry_streams/user_functions/function_template.py

```diff
@@ -0,0 +1,79 @@
+from abc import ABC, abstractmethod
+from typing import Any, Generic, Optional, TypeVar
+
+InputType = TypeVar("InputType")
+OutputType = TypeVar("OutputType")
+IntermediateType = TypeVar("IntermediateType")
+
+
+class AggregationBackend(ABC, Generic[OutputType]):
+    """
+    A storage backend that is meant to store windowed aggregates. Configurable
+    to the type of storage.
+    """
+
+    @abstractmethod
+    def flush_aggregate(self, aggregate: OutputType) -> None:
+        """
+        Flush a windowed aggregate to storage. Takes in the output from
+        the Accumulator.
+        """
+
+
+class Accumulator(ABC, Generic[InputType, IntermediateType, OutputType]):
+    """
+    The standard Accumulator template.
+    Define these functions to build a custom
+    Accumulator for aggregation.
+    """
+
+    def __init__(self, backend: Optional[AggregationBackend[OutputType]] = None):
+        self.backend = backend
```

Comment:
The fact that no method in this class reference the backend is a hint that the backend attribute should not be defined at this level and should be left to the subclasses. 
If having a constraint defined at this level was semantically useful, then the methods below would use it.

Since the interface of the accumulator does nothing useful for the backend, is the backend actually needed as an abstraction? OR is it an implementation detail of specific implementation of the accumulator ?